larcoreobj/SummaryData/GeometryConfigurationInfo.cxx

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file   larcoreobj/SummaryData/GeometryConfigurationInfo.cxx
+ * @brief  Description of the current configuration of detector geometry.
+ * @author Gianluca Petrillo (petrillo@slac.stanford.edu)
+ * @date   May 13, 2020
+ * @see    larcoreobj/SummaryData/GeometryConfigurationInfo.h
+ * 
+ * This is a header-only library.
+ */
+
+#include "larcoreobj/SummaryData/GeometryConfigurationInfo.h"
+
+// C/C++ standard library
+#include <ostream>
+
+
+// -----------------------------------------------------------------------------
+std::ostream& sumdata::operator<<
+  (std::ostream& out, sumdata::GeometryConfigurationInfo const& info)
+{
+  
+  if (info.isDataValid()) {
+    out <<   "Geometry information version: " << info.dataVersion;
+    
+    if (info.dataVersion >= sumdata::GeometryConfigurationInfo::DataVersion_t{1})
+    {
+      out
+        << "\nDetector name:               '" << info.detectorName << "'"
+        ;
+    } // version >= 1
+    
+    out
+      << "\nFull configuration:"
+      << "\n" << std::string(80, '-')
+      << "\n" << info.geometryServiceConfiguration
+      << "\n" << std::string(80, '-')
```

Comment:
There is a _cetlib_ facility for making horizontal rules.
```.cpp
#include "cetlib/HorizontalRule.h"

void print() 
{
  cet::HorizontalRule const rule{8}; // Number of repeated characters
  std::cout << rule('+') << '\n'  // prints "++++++++"
            << rule('-') << '\n'; // prints "--------"
}
```
Maybe overkill for what you need, though.

Reply:
I was not aware of it.
I will not adopt it in this case because I feel it's not worth the additional dependency — right now, the library is pure C++ and Standard Library.
(ok, the dictionary library pulls in a load of dependencies — MessageFacility?!? — so I guess the argument is not as strong as I thought)

Reply:
I agree there is no need to introduce the explicit dependency here.